util/vizerror/vizerror.go

```diff
@@ -42,3 +42,9 @@ func Wrap(err error) error {
 	}
 	return Error{err}
 }
+
+// As returns the first vizerror.Error in err's chain.
+func As(err error) (_ Error, ok bool) {
+	var e Error
+	return e, errors.As(err, &e)
```

Comment:
I don't immediately understand why this works. My intuition from the spec was that this shouldn't work, so I'm surprised. Or maybe this is undefined. I need to go read the spec more.

I thought the return statement assigned all its results in order, so it effectively compiled to:

```
      res[0] = e
      res[1] = errors.As(&e)
      return
```

In that case, the first result should always be the zero Error value. Yet it's not. Now I need to go read.

Maybe rewrite this to be more explicit and less tricky:

```go
	var e Error
        ok = errors.As(&e)
	return e, ok
```

Or name the result, and then it's also safe:

```go
func As(err error) (e Error, ok bool) {
	ok = errors.As(err, &e)
        return
```

cc @dsnet @creachadair for Go spec stuff

Reply:
e.g. https://go.dev/play/p/UHESK3ILu3j

Reply:
done.  (no more fixup! commits for me!)